### PR TITLE
A/B testing tab

### PR DESCRIPTION
**Pull request by kaedroho**

### CODE REVIEW

File: wagtail_ab_testing/wagtail_hooks.py [hunk 1]
```diff
@@ -49,6 +54,36 @@ def register_create_abtest_action_menu_item():
...
 
+# This is the only way to inject custom JS into the editor with knowledge of the page being edited
+class AbTestingTabActionMenuItem(ActionMenuItem):
+    def render_html(self, request, context):
+        if 'page' in context:
+            return format_html(
+                '<script src="{}"></script><script src="{}"></script><script>window.abTestingTabProps = JSON.parse("{}");</script>',
+                reverse('wagtail_ab_testing:javascript_catalog'),
+                versioned_static('wagtail_ab_testing/js/wagtail-ab-testing.js'),
+                escapejs(json.dumps({
```
Comment (KalobTaulien) [inline wagtail_ab_testing/wagtail_hooks.py:65]: Wow I didnt know `escapejs` was a thing! Nice!

File: wagtail_ab_testing/static_src/main.tsx [hunk 2]
```diff
@@ -36,4 +39,20 @@ document.addEventListener('DOMContentLoaded', () => {
             }
         });
     });
+
+    // A/B testing tab on page edito
+    if (abTestingTabProps) {
+        $('ul.tab-nav').append(`<li role="tab" aria-controls="tab-abtesting">
+            <a href="#tab-abtesting" class="">${gettext('A/B testing')}</a>
+        </li>`);
+        $('div.tab-content').append(`
```
Comment (KalobTaulien) [inline wagtail_ab_testing/static_src/main.tsx:48]: If we're loading jQuery into the final js code, I'm wondering if we could either make more use of it or to convert this to vanilla JS. Thoughts?

File: wagtail_ab_testing/compat.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import os
+
+if os.name == 'nt':
+    # Windows has a different strftime format for dates without leading 0
+    # https://stackoverflow.com/questions/904928/python-strftime-date-without-leading-0
```
Comment (KalobTaulien) [inline wagtail_ab_testing/compat.py:5]: This blew my mind.

**Review by KalobTaulien — APPROVED** [review]

File: wagtail_ab_testing/static_src/main.tsx (hunk 2 above)
Comment (kaedroho) [reply wagtail_ab_testing/static_src/main.tsx:48]: Yep, I agree with not using jQuery. This could probably be replaced with ``innerHTML = ``

**Pull request merged**